inlang/source-code/paraglide/paraglide-next/src/app/getLanguage.server.ts

```diff
@@ -9,9 +9,9 @@ import { PARAGLIDE_LANGUAGE_HEADER_NAME } from "./constants"
  * THIS WILL BECOME OBSOLETE ONCE WE FIGURE OUT HOW TO SET THE LANGUAGE BEFORE ANY NEXT CODE RUNS
  * Once that's the case we will be able to just use `languageTag()` instead
  */
-export function getLanguage<T extends string>(): T {
+export async function getLanguage<T extends string>(): Promise<T> {
```

Comment:
Can you change the client side getter function type to be sync again? 
 
The function has to be sync on the client. Otherwise, every message function (`m.message()`) is async, which requires wrapping every message function with an await (`await m.message()`) and wrapping components with suspense.  

The unit tests pass (and the type annotation make no difference to the runtime logic. i would keep it sync on the client though)

Reply:
@samuelstroschein done!